### PR TITLE
fix: comprehend deserialization of primitive types

### PR DESCRIPTION
**Pull request by jamesonwilliams**
Resolves: https://github.com/aws-amplify/amplify-android/issues/997

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by jamesonwilliams** [comment]
cc @fjnoyp

**Review by Amplifiyer — APPROVED** [review]

File: aws-api-appsync/src/main/java/com/amplifyframework/core/model/types/GsonJavaTypeAdapters.java [hunk 1]
```diff
@@ -101,13 +102,45 @@ public void write(JsonWriter jsonWriter, Class<?> clazz) throws IOException {
...
+                    case "float":
+                        return Float.class;
+                    case "double":
+                        return Double.class;
+                    case "char":
+                        return Character.class;
+                    case "void":
+                        return Void.class;
+                    default:
+                        throw new NoSuchElementException("No primitive with name = " + primitiveLabel);
```
Comment (Amplifiyer) [inline aws-api-appsync/src/main/java/com/amplifyframework/core/model/types/GsonJavaTypeAdapters.java:142]: nit: Probably an `IllegalArgumentException` is more appropriate than `NoSuchElementException`?

Comment (jamesonwilliams) [reply aws-api-appsync/src/main/java/com/amplifyframework/core/model/types/GsonJavaTypeAdapters.java:142]: I started with an `IllegalArgumentException`, and then changed it to this, for no particular reason. I'll move it back.

**Review by rjuliano — APPROVED** [review]
Ship it 🚢

**Pull request merged**